test/providers/cuda_helpers.cpp

```diff
@@ -202,23 +216,40 @@ int cuda_fill(CUcontext context, CUdevice device, void *ptr, size_t size,
         return -1;
     }
 
+    // set required context
+    CUcontext curr_context = nullptr;
+    set_context(context, &curr_context);
+
     int ret = 0;
     CUresult res =
         libcu_ops.cuMemsetD32((CUdeviceptr)ptr, *(unsigned int *)pattern,
                               size / sizeof(unsigned int));
     if (res != CUDA_SUCCESS) {
-        fprintf(stderr, "cuMemsetD32() failed!\n");
+        fprintf(stderr, "cuMemsetD32(%llu, %u, %zu) failed!\n",
+                (CUdeviceptr)ptr, *(unsigned int *)pattern,
+                size / pattern_size);
+        return -1;
+    }
+
+    res = libcu_ops.cuCtxSynchronize();
+    if (res != CUDA_SUCCESS) {
+        fprintf(stderr, "cuCtxSynchronize() failed!\n");
         return -1;
     }
 
+    // restore context
+    set_context(curr_context, &curr_context);
     return ret;
 }
 
-int cuda_copy(CUcontext context, CUdevice device, void *dst_ptr, void *src_ptr,
-              size_t size) {
-    (void)context;
+int cuda_copy(CUcontext context, CUdevice device, void *dst_ptr,
+              const void *src_ptr, size_t size) {
     (void)device;
 
+    // set required context
+    CUcontext curr_context = nullptr;
+    set_context(context, &curr_context);
```

Comment:
nit: Do we need to set a context? I though that `cuMemcpy` automatically detects the context of `dst_ptr` and `src_ptr`.

Reply:
test doesn't work without this (only on windows) - cuGetErrorString() returns "invalid argument"